File: transcendence/backend/common/search/CursorPageRequest.java
package transcendence.backend.common.search;

/**
 * 
 */
public class CursorPageRequest {
    private int pageSize; //number of items per page
    private String encodedCursor; //opaque string representing the position in the dataset, can be null for the first page

    public CursorPageRequest() {
        this.pageSize = 20;
        this.encodedCursor = null;
    }

    public CursorPageRequest(int pageSize, String encodedCursor) {
        this.pageSize = pageSize;
        this.encodedCursor = encodedCursor;
    }

    public int getPageSize() { return pageSize; }
    void setPageSize(int pageSize) { this.pageSize = pageSize; }
    public String getEncodedCursor() { return encodedCursor; }
    public void setEncodedCursor(String encodedCursor) { this.encodedCursor = encodedCursor; }
}

File: backend/src/Common/Services/Search/Model/CursorPageRequest.cs
namespace Trippie.Common.Services.Search.Model;

/// <summary>
/// Pagination request from the client.
/// </summary>
/// <param name="PageSize">
/// How many rows the client wants. Will be clamped to <c>[1, SearchOptions.MaxPageSize]</c>
/// by the compiler — a malicious client can't request 1,000,000 rows.
/// </param>
/// <param name="EncodedCursor">
/// Opaque cursor produced by a previous response. Null for the first page.
/// Internally it's just base64url(JSON of the last row's sort columns), but the client
/// MUST treat it as opaque — we reserve the right to change the encoding.
/// </param>
public sealed record CursorPageRequest(int PageSize = 20, string? EncodedCursor = null);
